great_expectations/datasource/new_datasource.py

```diff
@@ -444,6 +445,7 @@ def __init__(
         self._datasource_config.update(
             {"data_connectors": copy.deepcopy(data_connectors)}
         )
+        self._raw_config = raw_config or self._datasource_config
```

Comment:
If no `raw_config` is provided through the constructor, we leverage the default config.

In all supported workflows through the `DataContext`, we will have a `raw_config`. Only if the user decides to instantiate a standalone `Datasource` will they run into this situation.

Reply:
I thought better here to return an empty config or `None` since there is no raw config in that case. That way `raw_config` is safe from any unwanted substituted values.